components/clp-package-utils/clp_package_utils/scripts/compress_from_s3.py

```diff
@@ -0,0 +1,299 @@
+        help="The path (e.g. x.y) for the field containing the log event's timestamp.",
+    )
+    args_parser.add_argument(
+        "-t", "--tags", help="A comma-separated list of tags to apply to the compressed archives."
+    )
+    args_parser.add_argument(
+        "--no-progress-reporting", action="store_true", help="Disables progress reporting."
+    )
+
+    subparsers = args_parser.add_subparsers(dest="subcommand", required=True)
+
+    object_parser = subparsers.add_parser("s3-object", help="Compress specific S3 objects")
+    object_parser.add_argument("inputs", metavar="URL", nargs="*", help="S3 object URLs")
+    object_parser.add_argument(
+        "--inputs-from", type=str, help="File containing S3 object URLs (one per line)"
+    )
+
+    prefix_parser = subparsers.add_parser(
+        "s3-key-prefix", help="Compress all objects under S3 key prefix"
+    )
+    prefix_parser.add_argument("inputs", metavar="URL", nargs="*", help="S3 prefix URL")
+    prefix_parser.add_argument("--inputs-from", type=str, help="File containing S3 prefix URL")
+
+    parsed_args = args_parser.parse_args(argv[1:])
+    if parsed_args.verbose:
+        logger.setLevel(logging.DEBUG)
+    else:
+        logger.setLevel(logging.INFO)
+
+    try:
+        config_file_path = pathlib.Path(parsed_args.config)
+        clp_config = load_config_file(config_file_path, default_config_file_path, clp_home)
+        clp_config.validate_logs_dir()
+
+        validate_and_load_db_credentials_file(clp_config, clp_home, False)
+    except Exception:
+        logger.exception("Failed to load config.")
+        return -1
+
+    # Validate logs_input type is S3
+    if clp_config.logs_input.type != StorageType.S3:
+        logger.error(
+            "S3 compression requires logs_input.type to be '%s', but configured type is '%s'. "
+            "Please update your clp-config.yml.",
+            StorageType.S3,
+            clp_config.logs_input.type,
+        )
+        return -1
+
+    storage_engine: StorageEngine = clp_config.package.storage_engine
+    dataset = parsed_args.dataset
+
+    if StorageEngine.CLP_S == storage_engine:
+        dataset = CLP_DEFAULT_DATASET_NAME if dataset is None else dataset
+        try:
+            clp_db_connection_params = clp_config.database.get_clp_connection_params_and_type(True)
+            validate_dataset_name(clp_db_connection_params["table_prefix"], dataset)
+        except Exception as e:
+            logger.error(e)
+            return -1
+
+        if parsed_args.timestamp_key is None:
+            logger.warning(
+                "`--timestamp-key` not specified. Events will not have assigned timestamps and can "
+                "only be searched from the command line without a timestamp filter."
+            )
+    else:
+        logger.error(
+            f"S3 compression requires storage engine {StorageEngine.CLP_S}, "
+            f"but configured engine is {storage_engine}."
+        )
+        return -1
```

Comment:
```suggestion
    if StorageEngine.CLP_S != storage_engine:
        logger.error(
            f"S3 compression requires storage engine {StorageEngine.CLP_S}, but configured engine"
            f" is {storage_engine}."
        )
        return -1

    # TODO: The following dataset validation is duplicated in `compress.py`. We should extract it
    # into a common utility function.
    dataset = CLP_DEFAULT_DATASET_NAME if dataset is None else dataset
    try:
        clp_db_connection_params = clp_config.database.get_clp_connection_params_and_type(True)
        validate_dataset_name(clp_db_connection_params["table_prefix"], dataset)
    except Exception as e:
        logger.error(e)
        return -1

    if parsed_args.timestamp_key is None:
        logger.warning(
            "`--timestamp-key` not specified. Events will not have assigned timestamps and can "
            "only be searched from the command line without a timestamp filter."
        )
```
- We don't need if-else here, right?
- Adding a TODO to track code duplication.